week-14/homework/index.js

```diff
@@ -0,0 +1,157 @@
+//select all necessary elements
+let list = document.querySelector('[data-compenent="list"]');
+let loader = document.querySelector('[data-compenent="loader"]');
+let prevBtn = document.querySelector('[data-component="pagination:previous"]');
+let nextBtn = document.querySelector('[data-component="pagination:next"]');
+let selectPagination = document.querySelector('[data-component="pagination:select"]');
+let filter = document.querySelector('[data-compenent="filter"]');
+let currentPage = 1;
+let totalPages;
+let currentFilter = 'top';
+
+//renders the anime data
+let renderData = function (animeData) {
+    let data = animeData.data;
+
+    //loops through data and generates html
+    list.innerHTML = '';
+    for (let i = 0; i < data.length; i++) {
+        //check if title_english has null value
+        let title = data[i].title_english;
+        if (!title) {
+            title = data[i].title;
+        }
+        list.innerHTML += `
+        <div class="col-6 col-md-3 mb-5 text-center" title="${title}">
+          <a href="details.html?id=${data[i].mal_id}" class="link-primary d-block position-relative text-decoration-none" data-component="item">
+            <span class="position-absolute badge bg-danger top-0 end-0">
+                <i class="bi bi-star-fill"></i> ${data[i].score}
+            </span>
+            <span class="d-flex flex-column justify-content-center">
+                <img class="rounded shadow" src="${data[i].images.jpg.image_url}" data-component="image">
+                <span class="text-dark mt-2" data-component="title">${title}</span>
+            </span>
+          </a>
+        </div>`;
+    }
+
+    //renders the pagination selection
+    let page = animeData.pagination;
+    currentPage = page.current_page;
+    totalPages = page.items.count;
+
+    //clears page
+    selectPagination.innerHTML = '';
+    //loop through pages and generate html
+    for (i = 1; i <= totalPages; i++) {
+        if (i == currentPage) {
+            selectPagination.innerHTML += `<option selected value="${i}">${i}</option>`;
+        } else {
+            selectPagination.innerHTML += `<option value="${i}">${i}</option>`;
+        }
+    }
+
+    //hide loader and re-enable buttons when data received
+    loader.classList.add('d-none');
+    selectPagination.disabled = false;
+    nextBtn.disabled = false;
+    prevBtn.disabled = false;
+    filter.disabled = false;
+}
+
+let renderFilter = function (genresData) {
+    let allGenres = genresData.data;
```

Comment:
This whole function is too complex, if you fetch `genres` only once as I suggested below, then you will need to build this `select` content only once and it should be a lot simpler. Let's redo this one